### PR TITLE
Create StabilityPoolFactory

### PR DESCRIPTION
**Pull request by arthurka-o**

### CODE REVIEW

File: packages/contracts/contracts/Interfaces/IStabilityPoolFactory.sol [hunk 1]
```diff
@@ -0,0 +1,13 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.11;
+
+interface IStabilityPoolFactory {
+  function createNewSP(address _asset, address _stabilityPoolAddress) external;
+
+  function removeSP(address _asset) external;
+
+  function getAssetSP(address _asset) external view returns (address);
```
Comment (udosson) [inline packages/contracts/contracts/Interfaces/IStabilityPoolFactory.sol:10]: Can you change the function name to `getStabilityPoolByAsset`, please?
It is more describing

File: packages/contracts/contracts/StabilityPoolFactory.sol [hunk 2]
```diff
@@ -0,0 +1,24 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.11;
+
+import "./Dependencies/Ownable.sol";
+
```
Comment (udosson) [inline packages/contracts/contracts/StabilityPoolFactory.sol:5]: Let's import the stabilitypool interface here

File: packages/contracts/contracts/Interfaces/IStabilityPoolFactory.sol (hunk 1 above)
Comment (arthurka-o) [reply packages/contracts/contracts/Interfaces/IStabilityPoolFactory.sol:10]: Yeah, I had my reasoning for why I returned `address`, but I rechecked, and you're correct

**Review by udosson — APPROVED** [review]
LGTM!

**Review by udosson — APPROVED** [review]

**Pull request merged**